<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>786</width>
    <height>650</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>3D Scanning Software</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_4">
    <property name="spacing">
     <number>6</number>
    </property>
    <property name="leftMargin">
     <number>6</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>6</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <property name="spacing">
       <number>6</number>
      </property>
      <item>
       <widget class="QGroupBox" name="images_group">
        <property name="minimumSize">
         <size>
          <width>250</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>250</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="title">
         <string>Images</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
         <property name="spacing">
          <number>0</number>
         </property>
         <property name="leftMargin">
          <number>6</number>
         </property>
         <property name="topMargin">
          <number>6</number>
         </property>
         <property name="rightMargin">
          <number>6</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QTreeView" name="image_tree">
           <property name="selectionBehavior">
            <enum>QAbstractItemView::SelectRows</enum>
           </property>
           <property name="headerHidden">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <property name="spacing">
            <number>6</number>
           </property>
           <property name="topMargin">
            <number>6</number>
           </property>
           <property name="bottomMargin">
            <number>6</number>
           </property>
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>13</width>
               <height>17</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="select_all_button">
             <property name="maximumSize">
              <size>
               <width>56</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string>All</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="select_none_button">
             <property name="maximumSize">
              <size>
               <width>56</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string>None</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="change_dir_button">
             <property name="text">
              <string>Change dir...</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="current_image_group">
        <property name="title">
         <string>Current View</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <property name="spacing">
          <number>6</number>
         </property>
         <property name="margin">
          <number>6</number>
         </property>
         <item>
          <widget class="ImageLabel" name="image1_label" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
         <item>
          <widget class="ImageLabel" name="image2_label" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
         <item>
          <widget class="GLWidget" name="glwidget" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <property name="spacing">
       <number>6</number>
      </property>
      <item>
       <widget class="QGroupBox" name="actions_group">
        <property name="maximumSize">
         <size>
          <width>200</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="title">
         <string>Actions</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <property name="spacing">
          <number>6</number>
         </property>
         <property name="margin">
          <number>6</number>
         </property>
         <item>
          <widget class="QPushButton" name="capture_button">
           <property name="text">
            <string>Capture...</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="extract_corners_button">
           <property name="text">
            <string>Extract Corners</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="decode_button">
           <property name="text">
            <string>Decode</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="calibrate_button">
           <property name="text">
            <string>Calibrate</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="reconstruct_button">
           <property name="text">
            <string>Reconstruct</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer_2">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>476</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="display_group">
        <property name="maximumSize">
         <size>
          <width>200</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="title">
         <string>Display</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <property name="spacing">
          <number>6</number>
         </property>
         <property name="margin">
          <number>6</number>
         </property>
         <item>
          <widget class="QRadioButton" name="display_original_radio">
           <property name="text">
            <string>Original images</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="display_decoded_radio">
           <property name="text">
            <string>Pattern images</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="display_projector_radio">
           <property name="text">
            <string>Projector view</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="display_3dview_radio">
           <property name="text">
            <string>3D View</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="checkerboard_group">
        <property name="maximumSize">
         <size>
          <width>200</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="title">
         <string>Checkerboard</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <property name="margin">
          <number>6</number>
         </property>
         <property name="spacing">
          <number>6</number>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="corner_count_label">
           <property name="toolTip">
            <string>Interior number of corners</string>
           </property>
           <property name="text">
            <string>Corners</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="corners_size_label">
           <property name="text">
            <string>Size</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="corners_width_line">
           <property name="text">
            <string>1</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QLineEdit" name="corners_height_line">
           <property name="text">
            <string>1</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QSpinBox" name="corner_count_y_spin"/>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="corner_count_x_spin"/>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="robust_decode_group">
        <property name="maximumSize">
         <size>
          <width>200</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="title">
         <string>Decode</string>
        </property>
        <layout class="QFormLayout" name="formLayout_2">
         <property name="fieldGrowthPolicy">
          <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
         </property>
         <property name="labelAlignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
         <property name="horizontalSpacing">
          <number>6</number>
         </property>
         <property name="verticalSpacing">
          <number>6</number>
         </property>
         <property name="margin">
          <number>6</number>
         </property>
         <item row="0" column="0">
          <widget class="QPushButton" name="threshold_button">
           <property name="maximumSize">
            <size>
             <width>90</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="toolTip">
            <string>Apply threshold to the current pattern</string>
           </property>
           <property name="text">
            <string>Threshold</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="threshold_spin"/>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="b_label">
           <property name="toolTip">
            <string>Black light power ratio (projector)</string>
           </property>
           <property name="text">
            <string>b</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="b_line">
           <property name="toolTip">
            <string>Black light power ratio (projector)</string>
           </property>
           <property name="text">
            <string>0.0</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="m_label">
           <property name="toolTip">
            <string>Minimum direct light component</string>
           </property>
           <property name="text">
            <string>m</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QSpinBox" name="m_spin">
           <property name="toolTip">
            <string>Minimum direct light component</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="calibration_group">
        <property name="maximumSize">
         <size>
          <width>200</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="title">
         <string>Calibration</string>
        </property>
        <layout class="QFormLayout" name="formLayout">
         <property name="margin">
          <number>6</number>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="homography_window_label">
           <property name="toolTip">
            <string>Homography window size</string>
           </property>
           <property name="text">
            <string>H Win</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="homography_window_spin"/>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="reconstruction_group">
        <property name="maximumSize">
         <size>
          <width>200</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="title">
         <string>Reconstruction</string>
        </property>
        <layout class="QFormLayout" name="formLayout_5">
         <property name="horizontalSpacing">
          <number>6</number>
         </property>
         <property name="verticalSpacing">
          <number>6</number>
         </property>
         <property name="margin">
          <number>6</number>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="max_dist_label">
           <property name="text">
            <string>Max. ray distance</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="max_dist_line">
           <property name="text">
            <string>0.0</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QCheckBox" name="normals_check">
           <property name="toolTip">
            <string>Save point normals</string>
           </property>
           <property name="text">
            <string>Normals</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QCheckBox" name="colors_check">
           <property name="toolTip">
            <string>Save point colors</string>
           </property>
           <property name="text">
            <string>Colors</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QCheckBox" name="binary_file_check">
           <property name="toolTip">
            <string>Save pointcloud in binary format</string>
           </property>
           <property name="text">
            <string>Binary format</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>786</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="change_dir_action"/>
    <addaction name="save_vertical_image_action"/>
    <addaction name="save_horizontal_image_action"/>
    <addaction name="separator"/>
    <addaction name="quit_action"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="about_action"/>
   </widget>
   <widget class="QMenu" name="menuCalibration">
    <property name="title">
     <string>Calibration</string>
    </property>
    <addaction name="load_calibration_action"/>
    <addaction name="save_calibration_action"/>
    <addaction name="display_calibration_action"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuCalibration"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="change_dir_action">
   <property name="text">
    <string>Change dir...</string>
   </property>
  </action>
  <action name="quit_action">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="save_vertical_image_action">
   <property name="text">
    <string>Save vertical image...</string>
   </property>
  </action>
  <action name="about_action">
   <property name="text">
    <string>About...</string>
   </property>
  </action>
  <action name="save_horizontal_image_action">
   <property name="text">
    <string>Save horizontal image...</string>
   </property>
  </action>
  <action name="load_calibration_action">
   <property name="text">
    <string>Load...</string>
   </property>
  </action>
  <action name="save_calibration_action">
   <property name="text">
    <string>Save...</string>
   </property>
  </action>
  <action name="display_calibration_action">
   <property name="text">
    <string>Display...</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ImageLabel</class>
   <extends>QWidget</extends>
   <header>ImageLabel.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>GLWidget</class>
   <extends>QWidget</extends>
   <header>GLWidget.hpp</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
